source/server/hot_restarting_parent.cc

```diff
@@ -184,7 +195,11 @@ void HotRestartingParent::Internal::recordDynamics(HotRestartMessage::Reply::Sta
   }
 }
 
-void HotRestartingParent::Internal::drainListeners() { server_->drainListeners(); }
+void HotRestartingParent::Internal::drainListeners() {
+  Network::ExtraShutdownListenerOptions options;
+  options.non_dispatched_udp_packet_handler_ = *this;
```

Comment:
Is there a hot restarting test verifies that non_dispatched_udp_packet_handler_ is not nullopt?

Reply:
Updated a test to do this.